### PR TITLE
[READY]Helps address atmospherics nullspace runtimes

### PR DESCRIPTION
**Pull request by silicons**
woops
also makes it more debuggable.

### CODE REVIEW

File: code/game/machinery/pipe/construction.dm [hunk 1]
```diff
@@ -147,14 +147,16 @@ Buildable meters
 /obj/item/pipe/attack_self(mob/user)
 	return rotate()
 
+GLOBAL_LIST_INIT(pipe_construction_cache, list())
```
Comment (duncathan) [inline code/game/machinery/pipe/construction.dm]: put it on SSair imo

File: code/game/machinery/pipe/construction.dm [hunk 2]
```diff
@@ -147,14 +147,16 @@ Buildable meters
...
 /obj/item/pipe/proc/get_pipe_cache(type, direction)
-	var/static/list/obj/machinery/atmospherics/check_cache
+	var/list/obj/machinery/atmospherics/check_cache = GLOB.pipe_construction_cache
 	if(!islist(check_cache))
 		check_cache = list()
 	if(!check_cache[type])
 		check_cache[type] = list()
 	if(!check_cache[type]["[direction]"])
-		check_cache[type]["[direction]"] = new type(null, null, direction)
+		check_cache[type]["[direction]"] = new type(null, FALSE, direction)
```
Comment (duncathan) [inline code/game/machinery/pipe/construction.dm]: while you're at it you should give them an identifiable name, something along the lines of
```
var/obj/machinery/atmospherics/A = new type(null, FALSE, direction)
A.name = "(CACHE) [A.name]"
check_cache[type]["[direction]"] = A
```

**Review by duncathan — CHANGES_REQUESTED** [review]
please test to ensure this solves some of the issue. please also add some kind of debug feedback for HE pipes being instantiated in nullspace since those seem to be the culprit

File: code/game/machinery/pipe/construction.dm (hunk 2 above)
Comment (kevinz000) [reply code/game/machinery/pipe/construction.dm]: kk ok it!

**Comment by kevinz000** [comment]
tested, names added, subsystem stuff added. didn't runtime in testing.

File: code/game/machinery/pipe/construction.dm [hunk 3]
```diff
@@ -147,14 +147,16 @@ Buildable meters
 /obj/item/pipe/attack_self(mob/user)
 	return rotate()
 
-/obj/item/pipe/proc/get_pipe_cache(type, direction)
-	var/static/list/obj/machinery/atmospherics/check_cache
+/proc/get_pipe_construction_cache(type, direction)
```
Comment (Cyberboss) [inline code/game/machinery/pipe/construction.dm]: Why remove the specification?

Comment (duncathan) [reply code/game/machinery/pipe/construction.dm]: this does not need to go a global proc no

File: code/game/machinery/pipe/construction.dm [hunk 4]
```diff
@@ -147,14 +147,16 @@ Buildable meters
 /obj/item/pipe/attack_self(mob/user)
 	return rotate()
 
-/obj/item/pipe/proc/get_pipe_cache(type, direction)
-	var/static/list/obj/machinery/atmospherics/check_cache
+/proc/get_pipe_cache(type, direction)
```
Comment (duncathan) [inline code/game/machinery/pipe/construction.dm]: kevinz this is still a global proc please revert it to an `/obj/item/pipe` proc

**Review by duncathan — CHANGES_REQUESTED** [review]

**Pull request merged**